docs/guides/autoscaler-diskless-topics.md

```diff
@@ -0,0 +1,177 @@
+
+More information about how autoscaling for Diskless Topics works, and its benefits and limitations,
+is available in the [Aiven documentation](https://aiven.io/docs/products/diskless/howto/enable-autoscaling).
+
+## Enable service autoscaler
+
+The `autoscaler_service` feature cannot be enabled during service creation using the Aiven Terraform Provider. To enable the autoscaler,
+you first create the Kafka service with Diskless Topics enabled. After the service is created, you can use
+the `aiven_service_integration_endpoint` and `aiven_service_integration` resources to enable the `autoscaler_service` feature.
+
+1. Create a Kafka service. The following example creates a Kafka service with a `business-4` plan and enables Diskless Topics:
+
+      ```hcl
+      resource "aiven_kafka" "example_kafka" {
+        project      = var.project_name
+        cloud_name   = "google-europe-west1"
+        plan         = "business-4"
+        service_name = "example-kafka-diskless-staceys"
+        kafka_user_config {
+          kafka_diskless {
+            enabled = true
+          }
+        }
+      }
+      ```
+
+2. To create the service, run `terraform plan` and `terraform apply`.
+
+3. To let the autoscaler to adjust the plan, remove the plan attribute from the Kafka resource after the service is created.
+
+      ```hcl
+      resource "aiven_kafka" "example_kafka" {
+        project      = var.project_name
+        cloud_name   = "google-europe-west1"
+        service_name = "example-kafka-diskless-staceys"
+        kafka_user_config {
+          kafka_diskless {
+            enabled = true
+          }
+        }
+      }
+      ```
+
+4. To enable `autoscaler_service`, create a file with the autoscaler endpoint and service integration resources:
+
+      ```hcl
+      resource "aiven_service_integration_endpoint" "kafka_topic_autoscaler_endpoint" {
+        project       = var.project_name
+        endpoint_name = "topic-autoscaler"
+        endpoint_type = "autoscaler_service"
+      }
+
+      resource "aiven_service_integration" "diskless-topics-autoscaler" {
+        project             = var.project_name
+        integration_type    = "autoscaler_service"
+        source_service_name = aiven_kafka.example_kafka.service_name
+        dest_endpoint_id    = aiven_service_integration_endpoint.kafka_topic_autoscaler_endpoint.id
+        user_config = {
+          autoscaling = {
+            min_plan = "business-4"
+            max_plan = "business-32"
+          }
+        }
+      }
+      ```
+
+5. To preview the changes run `terraform plan`.
+
+6. Apply the changes by running `terraform apply`.
+
+You get an email notification whenever autoscaling scales the service up or down. You can also view the integration
+in the [Aiven Console](https://console.aiven.io) by going to the service and clicking **Integrations**.
```

Comment:
I would add a section with an output example. Once the `plan` field is removed, you can read it to know the current plan. It might be helpful to create alerts or dashboards with service health/billing.

```hcl
output "example_kafka_plan" {
  value = aiven_kafka.example_kafka.plan
}
```

```
terraform output example_kafka_plan
example_kafka_plan = "business-8"
```

https://developer.hashicorp.com/terraform/cli/commands/output